### DISCUSSION TITLE
Ability to disable bulk checkbox per row

### DISCUSSION THREAD

**Discussion by edwinvdpol** [rappasoft/laravel-livewire-tables#1907] [Feature Requests]
Is there also a possibility to determine per row if the bulk (checkbox) should be disabled?

Like, when I have a row, but the bulk actions cannot be executed on that row, I would like to disable (unchecked) the bulk checkbox.

Can something like this be done?

```php
class TableComponent extends DataTableComponent
{
    public function configure(): void
    {
        // ...
        $this->setBulkActionEnabledForRow(fn ($row) => $row->bulkEnabled));
        // or
        $this->setBulkActionDisabledForRow(fn ($row) => $row->bulkEnabled));
    }
}
```

**Comment by lrljoe** [comment]
Not at present!

**Comment by lrljoe** [comment]
So I am still pondering this one, and I think adding in a callback will be the approach, maybe October will be the time to implement this

**Reply by ahmadrivaldi-arv** [reply]
any update with this?

**Reply by lrljoe** [reply]
It's still on the to-do list, and not far down it either!

**Comment by lrljoe** [comment]
@edwinvdpol  @ahmadrivaldi-arv 

So - it'd be very simple to prevent the checkbox from appearing on a per-row basis with some callbacks.

However!  There's a need to re-validate the "selected items" when calling the Bulk Action, as a malicious end-user could maliciously add rows to the "selected items".  Which is where there's a slightly complexity.

Which means either: 
1) we introduce an "interim" solution that just hides the checkbox on a per-row basis, with a caveat that you need to do some validation
or
2) we introduce a version that actually handles it properly, which will be a bit longer in the works!

**Reply by ahmadrivaldi-arv** [reply]
I have tried adding a feature to add the ability to disable rows, tried it locally and it works fine

I will probably make a PR for this soon

**Reply by abrardev99** [reply]
> I have tried adding a feature to add the ability to disable rows, tried it locally and it works fine
> 
> I will probably make a PR for this soon

Mind sharing your solution?
